src/compas/datastructures/assembly/assembly.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import division
+
+from ..datastructure import Datastructure
+from ..network import Network
+
+
+class Assembly(Datastructure):
+
+    @property
+    def DATASCHEMA(self):
+        import schema
+        return schema.Schema({
+            "network": Network,
+        })
+
+    @property
+    def JSONSCHEMANAME(self):
+        return 'assembly'
+
+    def __init__(self, *args, **kwargs):
+        super(Assembly, None).__init__(*args, **kwargs)
+        self.attributes = {}
+        self.network = Network()
+
+    def __str__(self):
+        tpl = "<Assembly with {} elements and {} interactions>"
```

Comment:
part + connections works for me. sub-assemblies should not be a problem i think since the data structures can be nested to whatever level needed...

will add a few more things in the next few days, but want to keep this base class simple and then extend with more specialised versions later on.

Reply:
👍